src/lucky_record/model.cr

```diff
@@ -103,4 +107,89 @@ class LuckyRecord::Model
   macro association(table_name, foreign_key = nil)
     {% ASSOCIATIONS << {name: table_name.id, foreign_key: foreign_key} %}
   end
+
+  macro setup_correct_field_mappings_method
+    def ensure_correct_field_mappings!
+      ensure_existing_table!
+
+      fields = [
+        {% for field in FIELDS %}
+          { name: :{{field[:name]}}, nilable: {{ field[:nilable] }}, type: {{ field[:type] }} },
+        {% end %}
+      ]
+      ensure_fields_match_columns! fields
+    end
+  end
+
+  def ensure_existing_table!
+    table_names = LuckyRecord::Repo.tables_with_schema(excluding: "migrations")
+    return if table_names.includes?(@@table_name.to_s)
+
+    best_match = Levenshtein::Finder.find @@table_name.to_s, table_names, tolerance: 4
+    message = "The table #{@@table_name} was not found"
+
+    if best_match
+      message += ", did you mean #{best_match}?"
+    end
+
+    raise message
+  end
+
+  def ensure_fields_match_columns!(fields)
+    columns_map = Hash(String, Bool).new
+    LuckyRecord::Repo.table_columns(@@table_name).each do |column|
+      columns_map[column.name] = column.nilable
+    end
+
+    missing_columns = [] of String
+    optional_field_errors = [] of String
+    required_field_errors = [] of String
+
+    fields.each do |field|
+      unless columns_map.has_key? field[:name].to_s
+        missing_columns << missing_field_error(field)
+        next
+      end
+
+      if !field[:nilable] && columns_map[field[:name].to_s]
+        required_field_errors << required_field_error(field)
+      elsif field[:nilable] && !columns_map[field[:name].to_s]
+        optional_field_errors << optional_field_error(field)
+      end
+    end
+
+    if missing_columns.empty? && optional_field_errors.empty? &&required_field_errors.empty?
+      return
+    end
+
+    message = missing_columns + optional_field_errors + required_field_errors
+
+    raise message.join("\n\n")
+  end
+
+  private def missing_field_error(field)
+    "The table #{@@table_name} does not have a '#{field[:name]}' column. Make sure you've added it to the migration."
+  end
+
+  private def optional_field_error(field)
+    <<-ERROR
+    #{field[:name]} is marked as nilable (#{field[:name]} : #{field[:type]}?), but the database column does not allow nils.
+
+    Try this...
+
+      * Mark #{field[:name]} as non-nilable in your model: #{field[:name]} : #{field[:type]}
+      * Or, change the column in a migration to allow nils: make_optional :users, :#{field[:name]}
+    ERROR
+  end
+
+  private def required_field_error(field)
+    <<-ERROR
+    #{field[:name]} is marked as required (#{field[:name]} : #{field[:type]}), but the database column it not.
+
+    Try this...
+
+      * Mark #{field[:name]} as nilable in your model: #{field[:name]} : #{field[:type]}?
+      * Or, change the column in a migration to allow nils: make_optional :#{@@table_name}, :#{field[:name]}
```

Comment:
Does it make sense to recommend making the db column optional when the field is required?

Reply:
I think the message is intended to offer both possible solutions because the mistake could be either

Reply:
I'm still not sure I understand. `#{field[:name]} is marked as required`. Is allowing nils for the column really a possible solution? Wouldn't we need to make the db column not-nilable for this case?

Reply:
The user could have written the migrations first and marked a column `NOT NULL`, then made a mistake on the model field marking it as nilable`?`. Or the user might change the model without realizing the database column is still the same.

I think the error is intended to suggest solutions to both cases, or at least that's my understanding

Reply:
I think @hanneskaeufler is right. The second option should be `Or, change the column in a migration to be required: make_required: ...`. Since the column is already optional if it hits this case

Reply:
Oooooh, I completely missed your point @hanneskaeufler . I thought you meant that the message didn't need both `Mark 'field_name' as nilable...` and `Or, change the column...`. :man_facepalming: